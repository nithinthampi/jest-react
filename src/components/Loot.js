import React, { Component } from "react";
import { connect } from "react-redux";
import sinon from "sinon";
import { fetchBitcoin } from "../actions/bitcoin";

export class Loot extends Component {
  componentDidMount() {
    this.props.fetchBitcoin();
  }

  renderBitcoin() {
    const { bitcoin } = this.props;
    if (Object.keys(bitcoin).length === 0) return "";
    return (
      this.props.balance / parseInt(bitcoin.bpi.USD.rate.replace(",", ""), 10)
    );
  }

  render() {
    return (
      <div>
        <h2>Loot Balance: {this.renderBitcoin()}</h2>
      </div>
    );
  }
}

export default connect(state => state, { fetchBitcoin })(Loot);
