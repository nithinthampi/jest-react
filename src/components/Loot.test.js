import React from "react";
import { Loot } from "./Loot";
import { configure, shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Loot", () => {
  const fetchBitcoinMock = jest.fn();
  let props = { balance: 10, bitcoin: {}, fetchBitcoin: fetchBitcoinMock };
  let loot = shallow(<Loot {...props} />);
  it("renders correctly", () => {
    expect(loot).toMatchSnapshot();
  });

  describe("When the component mounts", () => {
    beforeEach(() => {
      loot = mount(<Loot {...props} />);
    });

    it("It dispatches the fetchBitcoin method", () => {
      expect(fetchBitcoinMock).toHaveBeenCalled();
    });
  });

  describe("When the component mounts completely", () => {
    beforeEach(() => {
      let props = {
        balance: 10,
        bitcoin: { bpi: { USD: { rate: "1,000" } } },
        fetchBitcoin: fetchBitcoinMock
      };
      loot = shallow(<Loot {...props} />);
    });

    it("renders the bitcoin amount", () => {
      expect(loot.find("h2").text()).toEqual("Loot Balance: 0.01");
    });
  });
});
