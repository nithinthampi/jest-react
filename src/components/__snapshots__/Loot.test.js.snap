// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loot
    balance={10}
    bitcoin={Object {}}
    fetchBitcoin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
        Loot Balance: 
        
</h2>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Loot Balance: ",
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Loot Balance: ",
        "",
      ],
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          Loot Balance: 
          
</h2>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Loot Balance: ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Loot Balance: ",
          "",
        ],
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
